### ISSUE TITLE
Focus outline of Radio checkbox should not be rectangle

### ISSUE THREAD

**Issue by huytd** [moaijs/moai#64]
The focus outline of radio button now displayed as rectangle, which should be in circle instead.

![image](https://user-images.githubusercontent.com/613943/111112267-1ce13880-851d-11eb-9667-ead63fd8f0a1.png)

**Comment by thien-do** [comment]
Yeah :( it's bad but I really don't know how to fix it yet.

Could use box-shadow but the thing is we are already using box-shadow for, well, shadow, and I don't want to use it for outline too because if there's something wrong it would be too subtle to catch.

PR are more than welcome

**Comment by ZeroX-DG** [comment]
According to the specs: https://www.w3.org/TR/CSS2/ui.html#propdef-outline
> the outline is the minimum outline that encloses all the element's boxes

So I don't think we have a "native" solution for this at the moment. You can either use box-shadow or a pseudo element :thinking:

**Comment by thien-do** [comment]
Is it gonna be changed? I read somewhere that in the future outline would respect the border-radius option. I hope it was not a dream.

> You can either use box-shadow or a pseudo element

I considered this option and I think we would not :( It's just too unnecessarily complicated, but the outcome is not paid off.

**Comment by huytd** [comment]
The specs said so:

> The parts of the outline are not required to be rectangular. To the extent that the outline follows the border edge, it should follow the border-radius curve.

But I don't think any browser implemented this yet.

Firefox does mentioned it in their doc, but still not working on my FF https://developer.mozilla.org/en-US/docs/Web/CSS/outline (see the last demo)

**Comment by huytd** [comment]
Some related issue trackers for references:
- https://bugs.chromium.org/p/chromium/issues/detail?id=81556
- https://bugs.webkit.org/show_bug.cgi?id=20807
- https://bugzilla.mozilla.org/show_bug.cgi?id=1031664

**Comment by thien-do** [comment]
Ok good news Firefox fixed this in 84!

Interestingly, using "outline-style: auto" in Safari also produces this behaviour, abeit it breaks our style. In other words, we still need to wait for webkit to fix it with outline-style: solid

The chromium bug also have activities, look like coming in 2021!
